### PR TITLE
 NeoBauer - solution for Ship and Cargo

### LINKED ISSUE #2: 0. Add Cargo and Ship classes from classes.
powinno być gotowe do 23:00 15.06.2020

### LINKED ISSUE #6: 1. Task: Cargo (operator==)
W klasie Cargo napisz operator porównania (operator==), który będzie sprawdzał, czy towary są takie same.

### PR DESCRIPTION
**Pull request by NeoBauer**

### CODE REVIEW

**Review by TomaszEichler — DISMISSED** [review]

**Review by mateusz-talma — CHANGES_REQUESTED** [review]
I think we should separate the class definitions from implementation. I would move definition to hpp file and in cpp file wrote methods.

**Comment by NeoBauer** [comment]
Added .hpp files - please check, if everything is fine.

File: shm/cargo.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#include <string>
+
```
Comment (mateusz-talma) [inline shm/cargo.cpp:4]: Please add #include "cargo.hpp"

File: shm/ship.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        , id_(id)
+    {}
+    Ship(int maxCrew, int speed, size_t id)
+        : Ship(0, maxCrew, speed, "", id)
+    {}
+
+    void setName(const std::string& name) { name_ = name; }
+
+    Ship& operator-=(size_t num) {
+        crew_ -= num;
```
Comment (mateusz-talma) [inline shm/ship.cpp:29]: crew_ can't be < 0

**Review by mateusz-talma — CHANGES_REQUESTED** [review]
Please add include hpp files in cpp

**Review by mateusz-talma — DISMISSED** [review]
Please refactor operator-= for screw

**Review by TomaszEichler — DISMISSED** [review]

File: shm/ship.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Ship::Ship(int maxCrew, int speed, size_t id)
+        : Ship(0, maxCrew, speed, "", id)
+    {}
+
+    void Ship::setName(const std::string& name) { name_ = name; }
+
+    Ship& Ship::operator-=(size_t num) {
+        if((crew_-num)<0){
+            crew_=0
+            return this;
```
Comment (mateusz-talma) [inline shm/ship.cpp]: You should return *this

Comment (mateusz-talma) [inline shm/ship.cpp]: You should return *this

File: shm/cargo.cpp [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#include <string>
+#include "cargo.hpp"
+
+    Cargo& Cargo::operator+=(size_t amount) {
+        amount_ += amount;
+        return *this;
+    }
+    Cargo& Cargo::operator-=(size_t amount) {
+        amount_ -= amount;
```
Comment (mateusz-talma) [inline shm/cargo.cpp:14]: amount_ < 0 ?

File: shm/cargo.hpp [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#include <string>
+#pragma once
```
Comment (mateusz-talma) [inline shm/cargo.hpp:1]: #pragma once should be placed before any headers are included.

File: shm/ship.cpp (hunk 2 above)
Comment (Marcinati) [reply shm/ship.cpp:29]: done

File: shm/cargo.cpp (hunk 4 above)
Comment (Marcinati) [reply shm/cargo.cpp:14]: done

**Review by Marcinati — DISMISSED** [review]
Everything is allrigth.

**Review by Marcinati — APPROVED** [review]

**Review by mateusz-talma — APPROVED** [review]

**Pull request merged**